backend/internal/storage/database.go

```diff
@@ -0,0 +1,55 @@
+package storage
+
+import (
+	"fmt"
+
+	"gorm.io/driver/postgres"
+	"gorm.io/gorm"
+)
+
+// Configuration to connect to database
+type DsnConfig struct {
+	// Username to login database
+	User string
+	// User password to verify user
+	Password string
+	// Database name to be accessed
+	DatabaseName string
+	// The host used to host database
+	Host string
+	// The port used to connect database
+	Port uint16
+}
+
+// Create a valid DSN string to connect database from DsnConfig
+func (config *DsnConfig) NewDsn() string {
+	dsn := fmt.Sprintf(
+		"user=%s password=%s dbname=%s host=%s port=%d",
+		config.User, config.Password, config.DatabaseName, config.Host, config.Port,
+	)
+	return dsn
+}
+
+// Connect to database by DsnConfig
+func ConnectDb(dsnConfig DsnConfig) (*gorm.DB, error) {
+
+	dsn := dsnConfig.NewDsn()
+
+	db, err := gorm.Open(postgres.Open(dsn), &gorm.Config{})
+	if err != nil {
+		return nil, fmt.Errorf("failed to connect to database: %w", err)
```

Comment:
Nice. This is the error formatting style you often see when reading Go code:
- lowercase
- some error message understandable from the context
- inner error appended to the end separated by a colon.

Nothing to change here.